src/monster.cpp

```diff
@@ -1578,12 +1592,17 @@ void monster::die(Creature* nkiller) {
     }
     dead = true;
     set_killer( nkiller );
+
     if( hp < -( type->size < MS_MEDIUM ? 1.5 : 3 ) * type->hp ) {
         explode(); // Explode them if it was big overkill
     }
     if (!no_extra_death_drops) {
         drop_items_on_death();
     }
+    Creature *kill;
+    if((kill=get_killer())!=nullptr){
```

Comment:
Standards: you should move that `get_killer()` to the line with `kill`. As in `Creature *killer = get_killer(); if( killer != nullptr ){`